cachi2/core/package_managers/gomod.py

```diff
@@ -0,0 +1,1030 @@
+import functools
+import fnmatch
+import logging
+import os
+import re
+import shutil
+import tempfile
+from datetime import datetime
+from pathlib import Path, PureWindowsPath
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import backoff
+import git
+import semver
+
+from cachi2.core.config import Config, get_worker_config
+from cachi2.core.errors import (
+    CachitoCalledProcessError,
+    FileAccessError,
+    InvalidFileFormat,
+    GoModError,
+    RepositoryAccessError,
+    UnsupportedFeature,
+    ValidationError,
+)
+from cachi2.core.paths import RequestBundleDir
+from cachi2.core.packages_data import PackagesData
+from cachi2.core.utils import run_cmd, load_json_stream
+
+
+logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+_run_gomod_cmd = functools.partial(run_cmd, exc_msg="Processing gomod dependencies failed")
+_MODULE_VERSION_RE = re.compile(r"/v\d+$")
+
+
+def fetch_gomod_source(request, dep_replacements=None, package_configs=None):
```

Comment:
Should dep_replacements and package_configs (or "packages"?) just be part of Request eventually?

Reply:
I believe the Request object should carry all the input parameters necessary for a single request. I made that change in the simplest way possible for now.